src/components/AudioDetail.vue

```diff
@@ -307,6 +328,127 @@ export default defineComponent({
       () => store.state.nowPlayingContinuously
     );
 
+    const pronunciationByPhrase = computed(() => {
+      let textArray: Array<string> = [];
+      accentPhrases.value?.forEach((accentPhrase) => {
+        let textString = "";
+        accentPhrase.moras.forEach((mora) => {
+          textString += mora.text;
+        });
+        if (accentPhrase.pauseMora) {
+          textString += "、";
+        }
+        textArray.push(textString);
+      });
+      return textArray;
+    });
+
+    store.watch(
```

Comment:
プロジェクト内ではstore.watchよりVue.watchが使われていそうでした。
（もしstore.watchを使うモチベーションがあれば知りたいです。）

---

Vue.watch seemed to be used more than store.watch in the project.
(If you have any motivation to use store.watch, I'd like to know.)

https://github.com/Hiroshiba/voicevox/blob/85bd39167e41be23449f724f81a4f3e04d7de342/src/components/MenuItem.vue#L87

Reply:
> Please rewrite this using computed.

I may get it wrong but I don't think there's a way to pass the `v-for` parameters `accentPhraseIndex` and `moraIndex` to a computed property... Could you give a further instruction, maybe with some example code?

---

> computedを使って書き直してください。

申し訳無いんですが、`v-for`にある `accentPhraseIndex` と `moraIndex`を`computed`に渡す方法がわかりません。もっと詳しい指示、もしくはサンプルコードがあれば助かります。

Reply:
I need to dynamically add new `audioKey` into the watch list, it involves with a nested structure of `watch()`, which as far as I know, can only be achieved by using `store.watch`.

---

新しい`audioKey`を監視リストに加えるため、2層の`watch`が必要です。それは私が知る限り`store.watch`しかできません。もしVueの`watch`だけでできる方法があれば教えてください。

Reply:
`computed`で、`accentPhraseIndex`と`moraIndex`を引数に持つ関数を返すと使えると思います。
もしくは`computed`を使わず、普通に関数を`return`したほうがわかりやすいかもしれません。

---

You can use `computed` to return a function with `accentPhraseIndex` and `moraIndex` as arguments.
Or, it may be easier to `return` the function instead of using `computed`.

Reply:
なるほど、`store.watch`を使う理由があるんですね。であれば問題ありません！
Okay, so there is a reason to use `store.watch`. Then, it''s no problem!